sdk/servicebus/azure-servicebus/azure/servicebus/management/aio/_management_client_async.py

```diff
@@ -0,0 +1,259 @@
+                        request_body, api_version=constants.API_VERSION)
+                )
+        except ValidationError:
+            # post-hoc try to give a somewhat-justifiable failure reason.
+            if isinstance(queue, (str, QueueDescription)):
+                raise_with_traceback(
+                    ValueError,
+                    message="queue must be a non-empty str or a QueueDescription with non-empty str queue_name")
+            raise_with_traceback(
+                TypeError,
+                message="queue must be a non-empty str or a QueueDescription with non-empty str queue_name")
+
+        return _convert_xml_to_object(queue_name, et, QueueDescription)  # type: ignore
+
+    async def update_queue(self, queue_description):
+        # type: (QueueDescription) -> QueueDescription
+        """Update a queue
+
+        :param queue_description: The properties of this `QueueDescription` will be applied to the queue in
+         ServiceBus. Only a portion of properties can be updated.
+         Refer to https://docs.microsoft.com/en-us/rest/api/servicebus/update-queue.
+        :type queue_description: ~azure.servicebus.management.QueueDescription
+        :returns: ~azure.servicebus.management.QueueDescription returned from ServiceBus.
+        """
+
+        if not isinstance(queue_description, QueueDescription):
+            raise TypeError("queue_description must be of type QueueDescription")
+
+        to_update = QueueDescription()
+
+        for attr in QUEUE_DESCRIPTION_SERIALIZE_ATTRIBUTES:
+            setattr(to_update, attr, getattr(queue_description, attr, None))
+        to_update.default_message_time_to_live = avoid_timedelta_overflow(to_update.default_message_time_to_live)
+        to_update.auto_delete_on_idle = avoid_timedelta_overflow(to_update.auto_delete_on_idle)
+
+        create_entity_body = CreateQueueBody(
+            content=CreateQueueBodyContent(
+                queue_description=to_update,
+            )
+        )
+        request_body = create_entity_body.serialize(is_xml=True)
+        with _handle_response_error():
+            try:
+                et = cast(
+                    ElementTree,
+                    await self._impl.queue.put(
+                    queue_description.queue_name,  # type: ignore
+                    request_body,
+                    api_version=constants.API_VERSION,
+                    if_match="*"
+                    )
+                )
+            except ValidationError:
+                # post-hoc try to give a somewhat-justifiable failure reason.
+                raise_with_traceback(
+                    ValueError,
+                    message="queue_description must be a QueueDescription with valid fields, "
+                            "including non-empty string queue name")
+
+        return _convert_xml_to_object(queue_description.queue_name, et, QueueDescription)  # type: ignore
+
+    async def delete_queue(self, queue_name):
```

Comment:
All of the APIs need to add `**kwargs` - as these include configuration that can be passed into the pipeline.
(So kwargs will need to be plumbed through to the generated APIs)

Reply:
This is a tangential comment, and I'm likely tilting at windmills, but the way we use kwargs such that they prevent ArgumentErrors from occuring if you e.g. typo something, always seemed like a pitfall to me.

Reply:
Added **kwargs to all APIs